### PR TITLE
Exposing power level overwrites on room creation to FFI

### PR DESCRIPTION
**Pull request by toger5**
This allows to overwrite the power levels on room creation. So the app can set the Power level for m.call.member events to 0.
This is an alternative to: https://github.com/ruma/ruma/pull/1706

- [x] Public API changes documented in changelogs (optional)

Signed-off-by:

### CODE REVIEW

**Review by bnjbvr — DISMISSED** [review]
This looks like a fairly Element-specific change, and not generic enough in that we're forcing something upon users of the create_room API, on which they have absolutely no control whatsoever.

Instead, could we include in the `CreateRoomParameters` 1. either a list of extra initial state events, and the apps would add the power level for call members? 2. or a boolean `allow_call_to_everyone` that would add the state event if set, and be unset by default?

**Comment by jplatte** [comment]
I agree with Benji. Specifically, the [room creation request](https://docs.ruma.io/ruma/api/client/room/create_room/v3/struct.Request.html) has a `power_level_content_override` field that can be used to set initial power levels. We should have a field like that in the FFI `CreateRoomParameters` type.

**Comment by bnjbvr** [comment]
@toger5 Thanks for updating; while it's a bit more work, can you look into Jonas' suggestion which sounds even better, please?

**Comment by toger5** [comment]
@jplatte @bnjbvr 
I am at: adding `power_level_content_override` to the ffi right now.

File: bindings/matrix-sdk-ffi/src/client.rs [hunk 1]
```diff
@@ -810,6 +812,85 @@ impl Client {
     }
 }
 
+#[derive(uniffi::Record)]
+pub struct NotificationPowerLevels {
+    pub room: i32,
+}
+impl From<NotificationPowerLevels> for ruma::power_levels::NotificationPowerLevels {
```
Comment (jplatte) [inline bindings/matrix-sdk-ffi/src/client.rs:819]:
```suggestion
}

impl From<NotificationPowerLevels> for ruma::power_levels::NotificationPowerLevels {
```

File: bindings/matrix-sdk-ffi/src/client.rs [hunk 2]
```diff
@@ -810,6 +812,85 @@ impl Client {
...
+                    None
+                }
+            })
+            .collect();
+
+        power_levels.events = value
+            .events
+            .iter()
+            .map(|(event_type, power_level)| {
+                let event_type: ruma::events::TimelineEventType = event_type.clone().into();
```
Comment (jplatte) [inline bindings/matrix-sdk-ffi/src/client.rs:885]:
```suggestion
                let event_type: ruma::events::TimelineEventType = event_type.as_str().into();
```

File: bindings/matrix-sdk-ffi/src/client.rs [hunk 3]
```diff
@@ -863,13 +944,17 @@ impl From<CreateRoomParameters> for create_room::v3::Request {
...
-        if value.allow_call_to_everyone {
-            let mut power_levels = RoomPowerLevelsEventContent::new();
-            power_levels.events = BTreeMap::from([(TimelineEventType::CallMember, int!(0))]);
-            initial_state.push(InitialStateEvent::new(power_levels).to_raw_any());
-        }
         request.initial_state = initial_state;
 
+        if let Some(power_levels) = value.power_level_content_override {
+            match Raw::new(&power_levels.into()) {
+                Ok(power_levels) => request.power_level_content_override = Some(power_levels),
```
Comment (jplatte) [inline bindings/matrix-sdk-ffi/src/client.rs:951]:
```suggestion
                Ok(power_levels) => {
                    request.power_level_content_override = Some(power_levels);
                }
```

**Review by jplatte — COMMENTED** [review]
Looks good apart from nitpicks.

**Review by jplatte — APPROVED** [review]

**Pull request merged**